### PR TITLE
JBPM-8903 : Stunner - Properties panel is not opened after clicking on "Minimize panel" button

### PR DESCRIPTION
**Pull request by inodeman**
Hi @romartin Can you review this PR?

### CODE REVIEW

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
@romartin @LuboTerifaj This is scoped for 7.7 - please **review**

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
@inodeman can you please generate the `kie-wb-distributions` WAR (`Business Central`) locally and upload it somewhere, so we can also test the UI? (meanwhile FDB are failing).
Thanks!

**Comment by inodeman** [comment]
War can be downloaded here
https://drive.google.com/file/d/1AydOQ7cxNnt396fFKFoi-DlvDrQL4y_8/view?usp=sharing

**Comment by inodeman** [comment]
Hi @romartin war was uploaded, can you review it?

**Comment by romartin** [comment]
thanks @inodeman . Anyway now FDB is green, so @LuboTerifaj can also start the review. Pllease guys, tthat's a priority for product. Thanks!

**Comment by LuboTerifaj** [comment]
Hello @inodeman , I have found one more issue.

Switch to full screen mode, close Properties panel and then repeat 3 times:
1. Maximize panel.
2. Hide alerts.
3. View alerts.
4. Minimize panel.
Animation that is not needed is being shown.

After that, execute just first two steps.
**The whole designer is shifted**. See the picture:
shifted-designer

**Comment by inodeman** [comment]
Hi @LuboTerifaj some of those issues I think happen with or without this PR, I think we should create Jira for it. Now for the Explore Panel I did not see it mentioned on the Ticket, also the behaviour about when the window is maximized and the user clicks properties panel and then on minimize does seem logical but was not mentioned previously. I think I can do the Explore Panel plus the behaviour of user clicking on Properties when Maximized in a few hours, but in case it takes more, I think could be a good idea to merge the PR and add the issues on Jiras, what do you think Lubo, @romartin ?

**Comment by inodeman** [comment]
@LuboTerifaj I saw the issue you comment about the panel being shifted, it is due to Uberfire and its docks it seems, although it only shows for about half a second and then it goes back to normal in my case. Now, I have seen from time to time that issue you comment where it just stays shifted. On your case is it reproducible 100% of the time? I mean that it stays shifted?

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Hi @LuboTerifaj I added the functionality you mentioned on the Explore Panel, so it should work now, about all other issues can we create other Jiras for them? let me know

**Comment by mbiarnes** [comment]
Jenkins execute full downstream build

**Comment by LuboTerifaj** [comment]
Hello @inodeman , can you apply kie code style formatting on your code please? Thanks!

**Comment by romartin** [comment]
@inodeman plz reformat code andl let us know agian. Thx!

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Hi @LuboTerifaj @romartin I re-checked the formatting and I have it set as KIE by default, changed a couple of conflicting issues, I think should be fine now, let me know

**Comment by romartin** [comment]
Jenkins retest this

**Comment by romartin** [comment]
ok @LuboTerifaj please missing your final review! thx.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/test/java/org/kie/workbench/common/stunner/client/widgets/palette/BS3PaletteWidgetImplTest.java [hunk 1]
```diff
@@ -172,7 +172,7 @@ public void checkDestructionReleasesResources() {
 
     @Test
     public void checkOnScreenMaximisedDiagramEditor() {
-        final ScreenMaximizedEvent event = new ScreenMaximizedEvent(true);
+        final ScreenMaximizedEvent event = new ScreenMaximizedEvent(true );
```
Comment (LuboTerifaj) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/test/java/org/kie/workbench/common/stunner/client/widgets/palette/BS3PaletteWidgetImplTest.java]:
```suggestion
        final ScreenMaximizedEvent event = new ScreenMaximizedEvent(true);
```

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-project-client/src/main/java/org/kie/workbench/common/stunner/bpmn/project/client/editor/BPMNDiagramEditor.java [hunk 2]
```diff
@@ -131,12 +138,53 @@ public String getEditorIdentifier() {
...
         String currentPerspectiveIdentifier = perspectiveManager.getCurrentPerspective().getIdentifier();
         Collection<UberfireDock> stunnerDocks = stunnerDocksHandler.provideDocks(currentPerspectiveIdentifier);
+
         stunnerDocks.stream()
-                .filter(dock -> dock.getPlaceRequest().getIdentifier().compareTo(DiagramEditorPropertiesScreen.SCREEN_ID) == 0)
-                .forEach(uberfireDocks::open);
-        super.doOpen();
+                .filter(dock -> dock.getPlaceRequest().getIdentifier().compareTo(id) == 0)
+                .forEach(action);
+
```
Comment (LuboTerifaj) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-project-client/src/main/java/org/kie/workbench/common/stunner/bpmn/project/client/editor/BPMNDiagramEditor.java]:
```suggestion
```

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/event/screen/ScreenPreMaximizedStateEvent.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * limitations under the License.
+ */
+
+package org.kie.workbench.common.stunner.core.client.event.screen;
+
+public class ScreenPreMaximizedStateEvent extends ScreenResizeEvent {
+
+    private final boolean isExplorerScreen;
+
+    public ScreenPreMaximizedStateEvent(final boolean isExplorerScreen ) {
```
Comment (LuboTerifaj) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/event/screen/ScreenPreMaximizedStateEvent.java]:
```suggestion
    public ScreenPreMaximizedStateEvent(final boolean isExplorerScreen) {
```

**Pull request merged**